services/sidecar/src/simcore_service_sidecar/mpi_lock.py

```diff
@@ -12,13 +12,19 @@
 import asyncio
 import datetime
 import logging
-from threading import Thread
+import multiprocessing
+import os
 from typing import Any, Callable, Optional, Tuple
 
 from aioredlock import Aioredlock, Lock, LockError
 
 from . import config
 
+# ptsv cause issues with multiprocessing
+# SEE: https://github.com/microsoft/ptvsd/issues/1443
+if os.environ.get("SC_BOOT_MODE") == "debug-ptvsd":
+    multiprocessing.set_start_method("spawn", True)
+
 logger = logging.getLogger(__name__)
 
 
```

Comment:
MINOR: Using ``tenacity`` in ``retry_for_result`` will improve readability

Reply:
code was refactored, an I nee to specifically handle LockError